Update array_ops.py

nikochiko opened the pull request:
Update documentation, formatting and fix typos for several methods.

nikochiko committed:
Update array_ops.py

Update documentation, formatting and fix typos for `tf.broadcast_dynamic_shape`, `tf.broadcast_static_shape`, `tf.boolean_mask`

nikochiko commented:
Batching fixes together. Will add commits on top of this.

nikochiko committed:
Update docstrings

Updated docstrings for `tf.convert_to_tensor` and `tf.edit_distance`.
`tf.convert_to_tensor`:  Put example in "For example:" section and switch to carets from backticks.
`tf.edit_distance`: Updated documentatoin, fixed example.

nikochiko committed:
Update save.py

Fix https://github.com/tensorflow/tensorflow/issues/34348 .
Notes:
- Documentation needs to be changed (in multiple places) after final changes in code.
- Changed code for deciding whether to save file as h5 or tf.
- Removed the unncessary _HDF5_EXTENSIONS list. Will have to make sure it wasn't used elsewhere.
- Added 4 new ValueError raises.

nikochiko committed:
Revert "Update save.py"

This reverts commit fdadd0e5e524df6488cd763c4ab7595d469ed1ef.

nikochiko committed:
Update docstring for tf.ensure_shape

Update docstring
Enhance example.

steph-en-m commented on tensorflow/python/ops/array_ops.py:
```diff
@@ -457,16 +465,23 @@ def broadcast_dynamic_shape(shape_x, shape_y):
+
+  >>> shape_x = tf.TensorShape([1, 2, 3])
+  >>> shape_y = tf.TensorShape([5, 1 ,3])
+  >>> broadcast_static_shape(shape_x, shape_y)
```
```>>> broadcast_static_shape(shape_x, shape_y)```

should be ```>>> tf.broadcast_static_shape(shape_x, shape_y)```

steph-en-m commented on tensorflow/python/ops/array_ops.py:
```diff
@@ -1630,9 +1636,21 @@ def boolean_mask_v2(tensor, mask, axis=None, name="boolean_mask"):
+
+  >>> tensor = [0, 1, 2, 3]  # 1-D example
+  >>> mask = np.array([True, False, True, False])
+  >>> boolean_mask(tensor, mask)
```
```>>> boolean_mask(tensor, mask)```
should be 
```>>> tf.boolean_mask(tensor, mask)```
Do the same to all parts where you missed calling ```tf.``` before the function names

nikochiko committed:
Make requested changes

steph-en-m commented on tensorflow/python/ops/array_ops.py:
```diff
@@ -3157,41 +3157,59 @@ def edit_distance(hypothesis, truth, normalize=True, name="edit_distance"):
+  For example:
+
+  Given the following input,
+  * `hypothesis` is a `tf.SparseTensor` of shape `[2, 1, 1]`
```
```...shape `[2, 1, 1]` ```
tensor shapes are tuples not lists. change every occurrence similar to the above to ```...shape `(2, 1, 1)` ```

steph-en-m commented on tensorflow/python/ops/check_ops.py:
```diff
@@ -2107,24 +2107,30 @@ def ensure_shape(x, shape, name=None):
-    # compatible with the shape (None, 3, 3)
-    sess.run(y, feed_dict={x: [1, 2, 3]})
+  >>> with tf.compat.v1.Session() as sess:
+  >>> sess.run(y, feed_dict={x: [1, 2, 3]})
```
Fix indentation here since `sess.run()` executes inside the `with` statement.

steph-en-m replied:
```
>>> with tf.compat.v1.Session() as sess:
...   sess.run(y, feed_dict={x: [1, 2, 3]})
```

nikochiko committed:
Make requested changes

- Shapes from lists to tuples
- TensorShapes to `TensorShape([...])`

nikochiko commented:
@steph-en-m Looking good?

steph-en-m commented:
Looks good to me. @k-w-w

nikochiko committed:
Minor changes

Made few more minor changes to documentation of `tf.extract_volume_patches` and `tf.fill`

nikochiko committed:
Merge branch 'master' into master

nikochiko commented:
Sorry, I had made this PR on `master`, so moved it to https://github.com/tensorflow/tensorflow/pull/34864. Please see.
